crates/stages/src/pipeline/ctrl.rs

```diff
@@ -15,11 +15,13 @@ pub(crate) enum ControlFlow {
         /// The progress of the last stage
         progress: BlockNumber,
     },
-    NoProgress,
+    NoProgress {
+        previous_progress: Option<BlockNumber>,
```

Comment:
I find the naming `progress` a bit confusing here, because this does not really convey what this precisely means.
is this the block number the previous stage is at? or the number of blocks the stage advanced/made progress? not really clear.

needs docs, can revise naming perhaps separately.

Reply:
this is the last progress of the current stage. agree on the renaming, added the docs to clarify